packages/config/src/options.js

```diff
@@ -448,6 +448,11 @@ export function getNuxtConfig (_options) {
       .map(([path, handler]) => ({ path, handler }))
   }
 
+  // Components module
+  if (options.components) {
+    options.buildModules.push('@nuxt/components')
```

Comment:
Side note: This is safe if user already specify in `modules` or `buildModules` section ([src](https://github.com/nuxt/nuxt.js/blob/dev/packages/core/src/module.js#L187)).